base/ca/src/main/java/com/netscape/cms/profile/constraint/P12ExportPasswordConstraint.java

```diff
@@ -0,0 +1,178 @@
+            if (matcher.find()) {
+                String passwordSubset = matcher.group(1);
+                catChars[0] = passwordSubset.split("(?<=\\p{Upper})", -6).length -1;
+                catChars[1] = passwordSubset.split("(?<=\\p{Lower})", -6).length -1;
+                catChars[2] = passwordSubset.split("(?<=\\d)", -6).length -1;
+                catChars[3] = passwordSubset.split("(?<=\\p{Punct})", -6).length -1;
+            }
+            for (int i = 0; i < 4; i++) {
+                if (catChars[i] < Integer.parseInt(catCharSize[i])) {
+                    throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                            "CMS_PROFILE_P12EXPORT_PASSWORD_FEW_CHARS_CATEGORY", catCharSize));
+                }
+            }
+        }
+
+        param = getConfig(CONFIG_PASSWORD_SUBSTRING_MATCH);
+        if (!param.equals("")) {
+            int seqSize = Integer.parseInt(param);
+            for (int i = 0; i< password.length() - (seqSize * 2); i++) {
+                String seq = password.substring(i, i + seqSize);
+                String invSeq = new StringBuilder(seq).reverse().toString();
+                if(password.indexOf(seq, i + seqSize) > 0 || password.indexOf(invSeq, i + seqSize) > 0){
+                    throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                            "CMS_PROFILE_P12EXPORT_PASSWORD_SEQUENCE"));
+                }
+            }
+        }
+
+        param = getConfig(CONFIG_PASSWORD_MAX_REPEATED_CHAR);
+        if (!param.equals("")) {
+            int maxRepeat = Integer.parseInt(param);
+            int mostRepeatedChar = password.chars().mapToObj(x -> (char) x)
+                    .collect(Collectors.groupingBy(x -> x, Collectors.counting()))
+                    .entrySet().stream().max(Map.Entry.comparingByValue())
+                    .get().getValue().intValue();
+
+            if (maxRepeat < mostRepeatedChar) {
+                throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                        "CMS_PROFILE_P12EXPORT_PASSWORD_REPEATED_CHAR", param));
+            }
+        }
+
+        if (getConfigBoolean(CONFIG_PASSWORD_CRACKLIB_CHECK)) {
+            try {
+                Process crack = new ProcessBuilder("/usr/sbin/cracklib-check").start();
+                BufferedWriter crackIn = crack.outputWriter();
+                BufferedReader crackOut = crack.inputReader();
+                crackIn.write(password);
+                crackIn.close();
+
+                String crackResult = crackOut.readLine().substring(password.length() + 2);
+                if (!crackResult.equals("OK")) {
+                    throw new ERejectException(CMS.getUserMessage(getLocale(req),
+                            "CMS_PROFILE_P12EXPORT_PASSWORD_CRACKLIB_FAILS", crackResult));
+                }
+
+            } catch (IOException e) {
+                logger.error(method + "impossible check password with cracklib.", e);
```

Comment:
If cracklib check is enabled and it's failing I think we should throw an exception here.